src/GitVersionCore/GitVersionCache.cs

```diff
@@ -43,56 +43,70 @@ public void WriteVariablesToDiskCache(IRepository repo, string gitDir, VersionVa
-                            vv = VersionVariables.FromFile(cacheFileName, fileSystem);
+                            fileSystem.Delete(cacheFileName);
                         }
-                        catch (Exception ex)
+                        catch (Exception deleteEx)
                         {
-                            Logger.WriteWarning("Unable to read cache file " + cacheFileName + ", deleting it.");
-                            Logger.WriteInfo(ex.ToString());
-                            try
-                            {
-                                fileSystem.Delete(cacheFileName);
-                            }
-                            catch (Exception deleteEx)
-                            {
-                                Logger.WriteWarning(string.Format("Unable to delete corrupted version cache file {0}. Got {1} exception.", cacheFileName, deleteEx.GetType().FullName));
-                            }
+                            Logger.WriteWarning(string.Format("Unable to delete corrupted version cache file {0}. Got {1} exception.", cacheFileName, deleteEx.GetType().FullName));
                         }
+
+                        return null;
                     }
                 }
-                else
-                {
-                    Logger.WriteInfo("Cache file " + cacheFileName + " not found.");
-                }
-
-                return vv;
             }
         }
 
-        string GetKey(IRepository repo, string gitDir)
+        string GetKey(GitPreparer gitPreparer)
         {
+            var dotGitDirectory = gitPreparer.GetDotGitDirectory();
+
             // Maybe using timestamp in .git/refs directory is enough?
-            var ticks = fileSystem.GetLastDirectoryWrite(Path.Combine(gitDir, "refs"));
-            var configPath = Path.Combine(repo.GetRepositoryDirectory(), "GitVersionConfig.yaml");
-            var configText = fileSystem.Exists(configPath) ? fileSystem.ReadAllText(configPath) : null;
-            var configHash = configText != null ? GetHash(configText) : null;
-            return string.Join(":", gitDir, repo.Head.CanonicalName, repo.Head.Tip.Sha, ticks, configHash);
+            var lastGitRefsChangedTicks = fileSystem.GetLastDirectoryWrite(Path.Combine(dotGitDirectory, "refs"));
+
+            // will return the same hash even when config file will be moved 
+            // from workingDirectory to rootProjectDirectory. It's OK. Config essentially is the same.
+            var configFilePath = ConfigurationProvider.SelectConfigFilePath(gitPreparer, fileSystem);
+            var configFileContent = fileSystem.Exists(configFilePath) ? fileSystem.ReadAllText(configFilePath) : null;
+            var configFileHash = configFileContent != null ? GetHash(configFileContent) : null;
+
+            return gitPreparer.WithRepository(repo => string.Join(":", dotGitDirectory, repo.Head.CanonicalName, repo.Head.Tip.Sha, lastGitRefsChangedTicks, configFileHash));
```

Comment:
If the value of the `/override` argument is the same, the output can be cached, no? If it can be cached, it should be cached, I think. That's why I believe the entire command line argument string should be made a part of the cache key, so whenever it changes in any way, the cache is invalidated.